pkg/cloudevents/event.go

```diff
@@ -31,7 +31,17 @@ func New(version ...string) Event {
 	return *e
 }
 
-// ExtensionAs returns Context.ExtensionAs(name, obj)
+// DEPRECATED: Access extensions directly via the e.Extensions() map.
```

Comment:
The plan in the SDK was not to just expose the fields, but the concept of an extension as a grouping of many keys in various places. 

I would like to not deprecate ExtensionAs, it is the flexible future.